### ISSUE TITLE
[quantization] Failed to save & reload quantized model

### ISSUE THREAD

**Issue by chenbohua3** [pytorch/pytorch#69426]
Given the codes:
```
import torch
import torchvision
from torch.fx import symbolic_trace
import torch.quantization.quantize_fx as quantize_fx

model = torchvision.models.resnet18().eval()
dummy = torch.randn(1, 3, 224, 224)
model = symbolic_trace(model)

qconfig_dict = {"": torch.quantization.get_default_qconfig('fbgemm')}
prepared_model = quantize_fx.prepare_fx(model, qconfig_dict)
prepared_model(dummy)
quant_module = quantize_fx.convert_fx(prepared_model)

torch.save(quant_module, 'test.pt')
torch.load('test.pt')
```

We will give:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/root/anaconda3/envs/test_torch110/lib/python3.6/site-packages/torch/serialization.py", line 607, in load
    return _load(opened_zipfile, map_location, pickle_module, **pickle_load_args)
  File "/root/anaconda3/envs/test_torch110/lib/python3.6/site-packages/torch/serialization.py", line 882, in _load
    result = unpickler.load()
  File "/root/anaconda3/envs/test_torch110/lib/python3.6/site-packages/torch/fx/graph_module.py", line 105, in reduce_graph_module
    return _deserialize_graph_module(forward, body)
  File "/root/anaconda3/envs/test_torch110/lib/python3.6/site-packages/torch/fx/graph_module.py", line 163, in _deserialize_graph_module
    graph = KeepModules().trace(com)
  File "/root/anaconda3/envs/test_torch110/lib/python3.6/site-packages/torch/fx/_symbolic_trace.py", line 560, in trace
    self.submodule_paths = {mod: name for name, mod in root.named_modules()}
  File "/root/anaconda3/envs/test_torch110/lib/python3.6/site-packages/torch/fx/_symbolic_trace.py", line 560, in <dictcomp>
    self.submodule_paths = {mod: name for name, mod in root.named_modules()}
  File "/root/anaconda3/envs/test_torch110/lib/python3.6/site-packages/torch/nn/modules/module.py", line 1695, in named_modules
    for m in module.named_modules(memo, submodule_prefix, remove_duplicate):
  File "/root/anaconda3/envs/test_torch110/lib/python3.6/site-packages/torch/nn/modules/module.py", line 1691, in named_modules
    for name, module in self._modules.items():
  File "/root/anaconda3/envs/test_torch110/lib/python3.6/site-packages/torch/nn/modules/module.py", line 1178, in __getattr__
    type(self).__name__, name))
AttributeError: 'ConvReLU2d' object has no attribute '_modules'
```

I tested the codes above with pytorch1.10.

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo

**Comment by vkuzo** [comment]
@chenbohua3 this is expected, the supported way to save and load quantized models is with the state dict.

cc @supriyar , we should make sure that this is explained in our documentation.

**Comment by supriyar** [comment]
Should be addressed by #69789
